### PR TITLE
#50 - AuthClientSlice added

### PR DESCRIPTION
**Pull request by olegmoz**
Part of #50 
`AuthClientSlice` added

### CODE REVIEW

File: src/test/java/com/artipie/http/client/auth/AuthClientSliceTest.java [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+        MatcherAssert.assertThat(
+            fake.capture(0),
+            new IsEqual<>(Headers.EMPTY)
+        );
+    }
+
+    @Test
+    void shouldAuthenticateOnceIfNotUnauthorized() {
+        final AtomicReference<Iterable<Map.Entry<String, String>>> capture;
+        capture = new AtomicReference<>();
```
Comment (olenagerasimova) [inline src/test/java/com/artipie/http/client/auth/AuthClientSliceTest.java:78]: @olegmoz looks like these two lines can be inlined

File: src/test/java/com/artipie/http/client/auth/AuthClientSliceTest.java [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+        MatcherAssert.assertThat(
+            fake.capture(1),
+            Matchers.containsInAnyOrder(rsheader)
+        );
+    }
+
+    @Test
+    void shouldAuthenticateTwiceIfNotUnauthorized() {
+        final AtomicReference<Iterable<Map.Entry<String, String>>> capture;
+        capture = new AtomicReference<>();
```
Comment (olenagerasimova) [inline src/test/java/com/artipie/http/client/auth/AuthClientSliceTest.java:126]: @olegmoz same here

**Review by olenagerasimova — CHANGES_REQUESTED** [review]
@olegmoz just cosmetic comments, take a look please

File: src/test/java/com/artipie/http/client/auth/AuthClientSliceTest.java (hunk 1 above)
Comment (olegmoz) [reply src/test/java/com/artipie/http/client/auth/AuthClientSliceTest.java:78]: @olenagerasimova I did it on purpose, line `final AtomicReference<Iterable<Map.Entry<String, String>>> capture = new AtomicReference<>();` is too long

Comment (olenagerasimova) [reply src/test/java/com/artipie/http/client/auth/AuthClientSliceTest.java:78]: @olegmoz hm, ok

**Review by olenagerasimova — APPROVED** [review]
@olegmoz good to merge

**Pull request merged**